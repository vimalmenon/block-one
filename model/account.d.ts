export interface AccountParams {
  customerId: string
  accountNo: string
}

export interface AccountModel {
  customerId: number
  accountNumber: string
  accountType: string
  balance: number
}
