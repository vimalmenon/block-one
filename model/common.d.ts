export type GenericMethodType<K> = (value: K) => void
export type GenericReturnMethodType<K, V> = (value: K) => V

export interface TableColumn {
  header: string
  accessor: string
  template?: any
}
