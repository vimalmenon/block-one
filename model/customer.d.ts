export interface CustomerParams {
  customerId: string
}

export interface CustomerModel {
  prefix: string
  name: string
  customerId: number
  address: string
  emailAddress: string
  contactNo: string
}
