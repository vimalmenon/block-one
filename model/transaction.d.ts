export interface TransactionModel {
  accountNumber: string
  timestamp: string
  action: string
  description: string
  amount: number
  currency: 'HKD' | 'USD'
}
