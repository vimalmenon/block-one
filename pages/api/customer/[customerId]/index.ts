import type { NextApiRequest, NextApiResponse } from 'next'

import { openDb } from '../../../../database.setup'

export default async (req: NextApiRequest, res: NextApiResponse<any>) => {
  if (req.method === 'GET') {
    const { customerId } = req.query
    const result = await (
      await openDb()
    ).all(
      'SELECT prefix, name, customer_id as customerId, address as address, email_address as emailAddress,  contact_no as contactNo FROM customers WHERE customer_id = ? LIMIT 1',
      customerId
    )
    res.status(200).json(result[0])
  }
}
