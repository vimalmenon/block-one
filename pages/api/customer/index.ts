import type { NextApiRequest, NextApiResponse } from 'next'
import { CustomerModel } from '../../../model/customer'

import { openDb } from '../../../database.setup'

export default async (req: NextApiRequest, res: NextApiResponse<CustomerModel[]>) => {
  if (req.method === 'GET') {
    const results = await (
      await openDb()
    ).all(
      'select prefix, name, customer_id as customerId, address, email_address as emailAddress, contact_no as contactNo from customers'
    )
    res.status(200).json(results)
  }
}
