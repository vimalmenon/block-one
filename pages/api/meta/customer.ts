import type { NextApiRequest, NextApiResponse } from 'next'
import { CustomerParams } from '../../../model/customer'

import { openDb } from '../../../database.setup'

export default async (req: NextApiRequest, res: NextApiResponse<CustomerParams[]>) => {
  if (req.method === 'GET') {
    const results = await (await openDb()).all('select customer_id as customerId FROM accounts')
    const params: CustomerParams[] = []
    results.forEach((result) => {
      params.push({
        customerId: result.customerId.toString(),
      })
    })
    res.status(200).json(params)
  }
}
