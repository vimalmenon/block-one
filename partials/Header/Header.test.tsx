import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { CustomerModel } from '../../model/customer';

import {Header} from "./";

describe("Header Partials", () => {
    test("Header : Snapshot testing without customer", () => {
        const wrapper = shallow(<Header/>);
        expect(toJson(wrapper)).toMatchSnapshot();
    });
    test("Header : Snapshot testing with customer", () => {
        const customer:CustomerModel = {
            prefix: "Mr",
            name: "test value",
            customerId:12345,
            address:"test Address",
            emailAddress:"test.value@test.com",
            contactNo:"123-456-789"
        };
        const wrapper = shallow(<Header customer={customer}/>);
        expect(toJson(wrapper)).toMatchSnapshot();
    });
});