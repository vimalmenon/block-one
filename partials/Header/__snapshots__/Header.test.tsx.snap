// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Partials Header : Snapshot testing with customer 1`] = `
<header
  className="flex h-28 items-center"
>
  <BusinessLogo />
  <div
    className="flex flex-grow self-center justify-between text-gray-700 font-semibold"
  >
    <div />
    <CustomerDetail
      customer={
        Object {
          "address": "test Address",
          "contactNo": "123-456-789",
          "customerId": 12345,
          "emailAddress": "test.value@test.com",
          "name": "test value",
          "prefix": "Mr",
        }
      }
    />
  </div>
</header>
`;

exports[`Header Partials Header : Snapshot testing without customer 1`] = `
<header
  className="flex h-28 items-center"
>
  <BusinessLogo />
  <div
    className="flex flex-grow self-center justify-center text-red-600"
  />
</header>
`;
